e2e/testcases/basic_test.go

```diff
@@ -0,0 +1,373 @@
+	nt.T.Log("Add an unmanaged resource into the namespace as a control")
+	nt.T.Log("We should never modify this resource")
+	_, err := nomostest.Retry(5*time.Second, func() error {
+		_, err := nt.Kubectl("apply", "-f", fmt.Sprintf("%s/reserved_namespaces/unmanaged-service.%s.yaml", yamlDir, namespace))
+		return err
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Add resource to manage")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/reserved_namespaces/service.yaml", yamlDir), fmt.Sprintf("acme/namespaces/%s/service.yaml", namespace))
+	nt.T.Log("Start managing the namespace")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/reserved_namespaces/namespace.%s.yaml", yamlDir, namespace), fmt.Sprintf("acme/namespaces/%s/namespace.yaml", namespace))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Start managing the namespace")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Wait until managed service appears on the cluster")
+	_, err = nomostest.Retry(30*time.Second, func() error {
+		_, err = nt.Kubectl("get", "services", "some-service", fmt.Sprintf("--namespace=%s", namespace))
+		return err
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Remove the namespace directory from the repo")
+	nt.RootRepos[configsync.RootSyncName].Remove(fmt.Sprintf("acme/namespaces/%s", namespace))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove the namespace from the managed set of namespaces")
+	nt.WaitForRepoSyncs()
+
+	nt.T.Log("Wait until the managed resource disappears from the cluster")
+	_, err = nomostest.Retry(60*time.Second, func() error {
+		_, err = nt.Kubectl("get", "services", "some-service", fmt.Sprintf("--namespace=%s", namespace))
+		return err
+	})
+	if err == nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.T.Log("Wait until the managed resource disappears from the cluster")
+	_, err = nt.Kubectl("get", "services", "some-other-service", fmt.Sprintf("--namespace=%s", namespace))
+	_, err = nomostest.Retry(600*time.Second, func() error {
+		_, err = nt.Kubectl("get", "services", "some-other-service", fmt.Sprintf("--namespace=%s", namespace))
+		return err
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	_, err = nomostest.Retry(60*time.Second, func() error {
+		_, err = nt.Kubectl("delete", "service", "some-other-service", "--ignore-not-found", fmt.Sprintf("--namespace=%s", namespace))
+		return err
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+}
+
+func unmanageNamespace(nt *nomostest.NT, namespace string) {
+	nt.T.Log("stop managing the system namespace")
+	nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/reserved_namespaces/unmanaged-namespace.%s.yaml", yamlDir, namespace), fmt.Sprintf("acme/namespaces/%s/namespace.yaml", namespace))
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Stop managing the namespace")
+	nt.WaitForRepoSyncs()
+}
+
+func TestNamespaceDefaultCanBeManaged(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation1)
```

Comment:
Are these intended to run for both monorepo and multirepo? The above tests all have the `SkipMultiRepo` flag.

Reply:
added skipmultirepo